src/huggingface_hub/repository.py

```diff
@@ -48,6 +48,41 @@ def is_local_clone(folder: Union[str, Path], remote_url: str):
     return remote_url in remotes
 
 
+def is_tracked_with_lfs(filename: str):
+    """
+    Check if the file passed is tracked with git-lfs.
+    """
+    folder = Path(filename).parent
+
+    try:
+        p = subprocess.run(
+            ["git", "check-attr", "-a", filename],
+            stderr=subprocess.PIPE,
+            stdout=subprocess.PIPE,
+            check=True,
+            encoding="utf-8",
+            cwd=folder,
+        )
+        attributes = p.stdout.strip()
+    except subprocess.CalledProcessError as exc:
+        if "not a git repository" in exc.stderr:
+            return False
+        else:
+            raise OSError(exc.stderr)
+
+    if len(attributes) == 0:
+        return False
+
+    found_lfs_tag = {"diff": False, "merge": False, "filter": False}
+
+    for attribute in attributes.split("\n"):
+        for tag in found_lfs_tag.keys():
+            if tag in attribute and "lfs" in attribute:
+                found_lfs_tag[tag] = True
+
+    return all(found_lfs_tag.values())
```

Comment:
if youre going through the hassle of parsing the output of `git check-attr` you can also just read `.gitattributes` and select all lines that end with `filter=lfs diff=lfs merge=lfs -text`

Reply:
That sounds a bit more complicated though, because I'd have to parse all wildcards to see if it contains the filename ... and then I'd have to do the same parsing; that sounds more complex than what's implemented here

Reply:
oh yes ok